api/src/utils.py

```diff
@@ -301,3 +301,42 @@ def add_date_to_dataset(path):
         "qualifies": [],
         "aliases": {},
     }
+
+
+def get_unique_items(list1, list2, id_property="_id"):
+    ids_list1 = set(item[id_property] for item in list1)
+    unique_items = [item for item in list2 if item[id_property] not in ids_list1]
+    return unique_items
+
+
+def alternate_lists(list1, list2, id_property=None):
+    """
+    Given to lists:
+    - a) ["k1", "k2", "k3", "k4", "k5", "k6", "k7", "k8"]
+    - b) ["s1", "s2", "s3", "s4", "s5", "s6", "s7", "s8"]
+
+    returns a new list of length max_lenght(a, b) and alternates items
+    stepping by 3. Example output:
+
+    ["k1", "k2", "k3", "s1", "s2", "s3", "k4", "k5", "k6", "s4"]
+
+    If id_property is provided, it is presumed the lists contain dictionaries,
+    and the duplicates are removed by checking the id_property in the dictionary
+    """
+
+    if id_property:
+        list2 = get_unique_items(list1, list2)
```

Comment:
TODO add id_property here, since this is working only because get_unique_items is defaulting to the same attribute (see its fn signature).

Reply:
Second optional TODO might be to also allow to remove duplicates when not a list of dicts. Not part of actual usage, but would help get_unique_items be more generic and align better with the tests I added.